lib/ReactGridLayout.jsx

```diff
@@ -152,11 +166,33 @@ var ReactGridLayout = React.createClass({
     this.setState({width: width});
   },
 
-  onDragStart(i, e, {element, position}) {
-    // nothing
+  /**
+   * When dragging starts
+   * @param {Number} i Index of the child
+   * @param {Number} x X position of the move
+   * @param {Number} y Y position of the move
+   * @param {Element} element The current dragging DOM element
+   * @param {Object} position Drag information
+   */
+  onDragStart(i, x, y, {e, element, position}) {
+    var layout = this.state.layout;
+    var l = utils.getLayoutItem(layout, i);
+    if (this.props.onDragStart(layout, this.state.layout, l, null, {e, element, position}) === false) {
```

Comment:
I like making it cancelable via the callback; any ideas how this is done in other libraries? My instinct is to check if the event has been canceled (e.g. e.preventDefault()) but that doesn't quite make as much sense in React.